### DISCUSSION TITLE
How can enemies be made to traverse uneven terrain?

### DISCUSSION THREAD

**Discussion by JimmyThrush** [Techokami/SonicWorldsNext#99] [Q&A] [answered]
![motobug](https://github.com/Techokami/SonicWorldsNext/assets/123826402/541fabab-29ff-4036-9f4e-3fc3b4afbb6e)

The motobug is able to clear the slopes on the right without issue, but is unable to climb any of the small, 1px ledges on the left like the player can. Is there a way to change this? I prefer the small changes in elevation this sort of tile arrangement creates for Sonic, but I'd like the enemies to be able to move across it as well.

For Sonic, if he's inside a top solid platform and sufficiently high enough, he'll be raised to the top of the platform, which is, to my understanding, how he's able to smoothly traverse this terrain (though he can still move across this tile arrangement if it's made of left/right solid platforms, albeit in a more choppy and unappealing fashion). How could this behavior be recreated for enemies?

**Comment by DimensionWarped** [comment, marked as answer]
You might be able to fix this just by using a different collision mesh. The issue here is that the motobug checks for whether or not it's touching a wall using 'is_on_wall' to decide when to turn around and the motobug CollisionMask is hitting a wall when you have these big steps to traverse. You could also split his collision into a foot collision and a side collision, raise the side collision mask a little higher, and check *that* for deciding whether or not motobug has hit a wall... but that's probably more work.

**Reply by JimmyThrush** [reply]
Thanks, changing the collision mesh worked. I probably should've figured that out on my own... but I appreciate the assistance.

**Reply by DimensionWarped** [reply]
No problem. Glad I could help.
